File: cpp/sorting/assign_cookies.cpp
//
// Created by Harsha Vardhan on 06/10/22.
//

#include <iostream>
#include <vector>
#include <unordered_map>

using namespace std;

int findContentChildren(vector<int> &g, vector<int> &s) {
    sort(g.begin(), g.end());
    sort(s.begin(), s.end());

    int i = 0, j = 0, ans = 0;
    while (i < g.size() && j < s.size()) {
        if (s[j] >= g[i]) {
            ans++;
            i++;
            j++;
        } else {
            j++;
        }
    }
    return ans;
}

int main() {
    return 0;
}

File: go/greedy/assign_cookies.go
package main

import (
	"fmt"
	"sort"
)

func findContentChildren(g []int, s []int) int {
	sort.Slice(g, func(i, j int) bool {
		return g[i] < g[j]
	})
	sort.Slice(s, func(i, j int) bool {
		return s[i] < s[j]
	})

	ans := 0
	i, j := 0, 0

	for i < len(s) && j < len(g) {
		if s[i] >= g[j] {
			ans++
			i++
			j++
		} else {
			i++
		}
	}
	return ans
}

func main() {
	fmt.Println(findContentChildren([]int{7, 8, 9, 10}, []int{5, 6, 7, 8}))
}
